AlfrescoKit/AlfrescoKit/Picking View Controller/AKAlfrescoNodePickingListViewController.m

```diff
@@ -0,0 +1,258 @@
+                                              delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                               session:(id<AlfrescoSession>)session
+{
+    return [self initAlfrescoFolderPickerWithRootFolder:folder selectedNodes:selectedNodes delegate:delegate listingContext:nil session:session];
+}
+
+- (instancetype)initAlfrescoFolderPickerWithRootFolder:(AlfrescoFolder *)folder
+                                         selectedNodes:(NSMutableArray *)selectedNodes
+                                              delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                        listingContext:(AlfrescoListingContext *)listingContext
+                                               session:(id<AlfrescoSession>)session
+{
+    self = [self initWithAlfrescoFolder:folder listingContext:listingContext session:session];
+    if (self)
+    {
+        self.nodePickerOptions = AKAlfrescoNodePickerTypeFolderPicker;
+        self.delegate = delegate;
+        self.selectedNodes = (selectedNodes) ? selectedNodes : [NSMutableArray array];
+    }
+    return self;
+}
+
+- (instancetype)initAlfrescoDocumentPickerWithRootFolder:(AlfrescoFolder *)folder
+                                       multipleSelection:(BOOL)allowMultiple
+                                           selectedNodes:(NSMutableArray *)selectedNodes
+                                                delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                                 session:(id<AlfrescoSession>)session
+{
+    return [self initAlfrescoDocumentPickerWithRootFolder:folder multipleSelection:allowMultiple selectedNodes:selectedNodes delegate:delegate listingContext:nil session:session];
+}
+
+- (instancetype)initAlfrescoDocumentPickerWithRootFolder:(AlfrescoFolder *)folder
+                                       multipleSelection:(BOOL)allowMultiple
+                                           selectedNodes:(NSMutableArray *)selectedNodes
+                                                delegate:(id<AKAlfrescoNodePickingListViewControllerDelegate>)delegate
+                                          listingContext:(AlfrescoListingContext *)listingContext
+                                                 session:(id<AlfrescoSession>)session
+{
+    self = [self initWithAlfrescoFolder:folder listingContext:listingContext session:session];
+    if (self)
+    {
+        self.nodePickerOptions = AKAlfrescoNodePickerTypeDocumentPicker;
+        if (allowMultiple)
+        {
+            self.nodePickerOptions |= AKAlfrescoNodePickerTypeMultiplePicker;
+        }
+        self.delegate = delegate;
+        self.selectedNodes = (selectedNodes) ? selectedNodes : [NSMutableArray array];
+        
+    }
+    return self;
+}
+
+- (NSString *)nibName
+{
+    return NSStringFromClass([self superclass]);
```

Comment:
As-is, the superclass is `AKAlfrescoNodeListViewController`. I'm not sure myself, but if a subclass of this class is created, will `self` then be the subclass and hence `[self superclass]` become `AKAlfrescoNodePickingListViewController`. If that's not the case, then ignore me -- maybe it's only an issue if the `nibName` method is overridden in the subclass.